tests/plot/test_plot.py

```diff
@@ -19,9 +21,39 @@
 
 
 class TestPlot(BaseTestCase):
-
     """Test Plot module"""
 
+    @pytest.mark.localonly
+    def test_generate_plot_animated(self):
+        """Test generate_plot function."""
+        filename = "tests/plot/test_generate_plot.gif"
+
+        # remove the file first
+        try:
+            pl.Path(filename).unlink()
+        except FileNotFoundError:
+            pass
+
+        numpoints = 100
+        xs = list(range(0, numpoints))
+        ys = random.choices(list(range(0, 1000)), k=numpoints)
+        ys2 = random.choices(list(range(0, 1500)), k=numpoints)
+
+        generate_plot(
+            [xs, xs],
```

Comment:
[xs1, xs2] is it possible that this list contains items with differing lengths. can xs1 and xs2 have different lengths ?

Reply:
Yes, I think they can, but [y1, y2] must have the same lengths as [x1, x2] so that matplotlib has complete (x,y) co-ordinates for each point. 

For animation, one will have to find the values with the longest length and use that to calculate duration and all that then?

Reply:
>> "For animation, one will have to find the values with the longest length and use that to calculate duration and all that then?"

Yeah that will be correct as, [plt1, plt2] are individual plots and we only care of the total duration which means the max of the lengths